Overhauled handling of compiler flags. Closes #8.

12qu opened the pull request:
Previously all compiler flags (cxxflags etc) were hardcoded into `ino',
which is inflexible since, for instance, it is not even possible to
define constants (via `-D') on the command-line at compile time.  This
commit adds the `--cppflags', `--cflags', `--cxxflags' and `--ldflags'
command-line options in order to improve this situation.

In summary, all are documented under `ino build --help'; defaults are
preserved as before; and the user is not able to void any flags (e.g.
include paths for the required libraries) which are essential for the
building of a sketch

jrmcornish committed:
Overhauled handling of compiler flags

Previously all compiler flags (cxxflags etc) were hardcoded into `ino',
which is inflexible since, for instance, it is not even possible to
define constants (via `-D') on the command-line at compile time.  This
commit adds the `--cppflags', `--cflags', `--cxxflags' and `--ldflags'
command-line options in order to improve this situation.

In summary, all are documented under `ino build --help'; defaults are
preserved as before; and the user is not able to void any flags (e.g.
include paths for the required libraries) which are essential for the
building of a sketch

12qu commented:
Note that this closes #8 since `-Wall' can now be turned on and off as desired by the user.

jrmcornish committed:
Fixed hanging when no .cpp files bug

Fixed bug that caused `ino' to hang when scanning for dependencies of
libraries with only .h files (i.e. no .ino, .c or .cpp files).

jrmcornish committed:
Allowed specification of build tools

Previously the verisons of avr-gcc etc. that were used for compilation
and linking were all set by default. This commit adds the option to
choose which tools to use via various flags specified in `ino build
--help'.

michaelsproul commented:
@12qu This looks sensible. Have you heard anything from the project maintainer? (I also have a pull request I'd like merged, but it seems I may never get a response)

jrmcornish committed:
Minor housekeeping

Made makefile select 'python2' via env for portability. Added vim
.sw[op] files to .gitignore. Added a TODO list.

jrmcornish committed:
Moved all build flag arg parsing into build.py

It is unnecessary to clutter environment.py since we don't require these
flags globally.

12qu commented:
@gnusouth Unfortunately no, I haven't heard anything yet. I am hopeful that this repo hasn't gone completely cold, but in the meantime feel free to send your changes via my repo and I will merge them. (I had a look at your last pull request and it looks good to me.)

I'm using ino quite a bit at the moment and am adding features and maintaining it as I go to fit my needs, and I'll hopefully be a bit more responsive than here.
